runtime/vm/jnicsup.cpp

```diff
@@ -498,31 +507,23 @@ UDATA JNICALL   pushArguments(J9VMThread *vmThread, J9Method* method, void *args
 				*(I_32*)(--sp) = (I_32)(jint)ARG(jint, i);	/* NOTE: jint may be larger than int, but jchar, jshort, etc, must be no larger than int */
 				break;
 #ifdef J9VM_INTERP_FLOAT_SUPPORT
-			case 'F':
+			case 'F': {
 				/* float type */
-				fltPtr = (jfloat*)--sp;
-				/* The Linux SH4 compiler needs the next two lines to be two steps.  If you combine them it fails to compile */
-				dbl = ARG(jdouble, f);
-				*fltPtr = (jfloat)dbl;
-				break;
+				sp -= 1;
+				/* jfloat {aka float}' is promoted to 'double' when passed through '...' */
+				*((jfloat*) sp) = (jfloat) ARG(jdouble, f);
+			}
+			break;
 			case 'D':
 				/* double type */
-				dbl = ARG(jdouble, d);
-				lngOrDblPtr = (UDATA *) &dbl;
-				goto pushLongOrDouble;
+				sp -= 2;
+				*((jdouble*) sp) = (jdouble) ARG(jdouble, d);
```

Comment:
No need to cast ARG.

Reply:
Done. I can remove the cast from 'B', 'Z', et al. and fix the `--sp` too, if you wish.

Reply:
I believe the casts for smaller-than-int cases are necessary (unless we completely trust the compiler/varargs implementation). Safer to ensure that we are properly truncating the int value and then sign/zero extending it before copying to the stack.